lib/fluent/plugin/in_tail.rb

```diff
@@ -948,18 +950,27 @@ def limit_bytes_per_second_reached?(start_reading, number_bytes_read)
           false
         end
 
+        def refresh_bytes_read_counter
+          if Fluent::Clock.now - @last_read_time > 1
+            @number_bytes_read = 0
+          end
+
+          yield
+        end
+
         def handle_notify
           with_io do |io|
```

Comment:
Should return immediately when we are already encountering the read limit (until passes 1 sec).

Reply:
With previous implementation, your suggestion shouldn't be handled here.
But, current patch https://github.com/fluent/fluentd/pull/3364/commits/34fcda1d69f9692473a8ad0ac9b2c8cbffc96357 should handle here due to stored time with `@start_reading`.

Reply:
Hmm, hard to tell my intention by words.
I've added a commit to simplify it.